Add duwi integration

duwi2024 opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
By summarizing the latest enhancements and underlying SDK updates, this proposed change underscores our commitment to improving home automation through continuous innovation. We believe these developments strengthen our offering, making a compelling case for the inclusion of this updated Duwi Integration in the Home Assistant platform.


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/33089


## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. Your PR cannot be merged unless tests pass
- [ ] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (ruff format homeassistant tests)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [x] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [x] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

Ledgerbiggg committed:
feat(duwi): Add Duwi integration

- Add support for Duwi devices in Home Assistant.
- Ensure compatibility and testing completed.

Ledgerbiggg committed:
Add a new integration of hub type: duwi

Ledgerbiggg committed:
Merge branch 'dev' into add-duwi-integration

duwi2024 committed:
Merge pull request #4 from Ledgerbiggg/add-duwi-integration

Add duwi integration

Ledgerbiggg committed:
Add the version number of the new integrated duwi

duwi2024 committed:
Merge pull request #6 from Ledgerbiggg/add-duwi-integration

Add the version number of the new integrated duwi

duwi2024 committed:
Merge branch 'dev' into add-duwi-integration

frenck reviewed (changes requested):
This PR contains three platforms (light, cover, switch). Please limit the initial addition of a new integration to a single platform. Additional platforms can be added in follow up PRs.

../Frenck

Ledgerbiggg committed:
Merge remote-tracking branch 'origin/dev' into dev

Ledgerbiggg committed:
chore(duwi): Limit initial integration to single platform

Following Home Assistant's contribution guidelines, this commit limits the initial addition of the new duwi integration to the switch platform only. Additional platforms (cover, light) to be added in subsequent PRs as recommended.

duwi2024 committed:
Merge branch 'pr/n7_add-duwi-integration' into add-duwi-integration

# Conflicts:
#	homeassistant/components/duwi/const.py
#	homeassistant/components/duwi/cover.py
#	homeassistant/components/duwi/light.py
#	homeassistant/components/duwi/manifest.json

Ledgerbiggg committed:
chore(duwi): Remove Cover and Light from duwi integration

duwi2024 committed:
Merge pull request #8 from Ledgerbiggg/add-duwi-integration

chore(duwi): Remove Cover and Light from duwi integration

duwi2024 committed:
Delete files

duwi2024 committed:
Merge branch 'dev' into add-duwi-integration

duwi2024 committed:
Merge branch 'dev' into add-duwi-integration

Ledgerbiggg commented:
@frenck I've made the requested changes. Could you please take another look when you have a moment? Thank you!

duwi2024 committed:
Merge branch 'dev' into add-duwi-integration

duwi2024 committed:
Merge branch 'dev' into add-duwi-integration

joostlek reviewed (changes requested):
Apparently there is a merge conflict, can you fix that?

emontnemery commented:
@duwi2024 Please update the PR description to explain what functionality is added to Home Assistant. The current explanation seems like the rationale for Duwi investing time in a Home Assistant integration, which is not relevant here.

Maybe something like:
> Add support for Duwi switches <link to product here>

Also, while I can find the PyPi package <https://pypi.org/project/duwi-smarthome-sdk/>, I can't find a repository with source code. Can you please update the PyPi page with a link to the source repo?
